components/cam/src/physics/cam/phys_grid.F90

```diff
@@ -6457,9 +6457,9 @@ subroutine proc_heap_adjust(updated, remove, heap_dlen, &
    integer :: proc_last    ! process id of last element
    integer :: proc_updated ! process id of updated element
    integer :: proc_i       ! process id of current element
-   real(r8):: proc_ip      ! process id of parent of current element
-   real(r8):: proc_il      ! process id of left child of current element
-   real(r8):: proc_ir      ! process id of right child of current
+   integer :: proc_ip      ! process id of parent of current element
```

Comment:
Hmm...thinking if it's better to move non-machine file kind of changes to a different PR. Is this a specific bugfix for an issue?

Reply:
To clarify, I do think integer datatype is a better fit here.

Reply:
All changes here are Cray-specific: this one too, in that Intel and GNU compilers ignore this issue, but Cray reports compile error.

Reply:
I'll put this in my bug fix PR for this routine, as this is a bug (if innocuous).